canonical/ubuntu.com
# Microcerts refactor

## Done

- Refactored microcerts views and UA contracts API to use the advantage decorator
- Changed to be mergeable to main branch
- Fixed tests and updated cassettes
- Re-refactored to make it work with both old view and new React table (hence the duplication of code)

## QA

Everything should work as it was working before

- Check that https://ubuntu-com-10631.demos.haus/cube/microcerts.json?test_backend=true is showing all the data (this works with both old view and new React table)
- Purchase


1. POST https://ubuntu-com-10631.demos.haus/cube/microcerts/purchase.json?test_backend=true
```
{
  "preview": "true", # You need to do first a preview and then do it a preview: "false" to complete the purchase
  "account_id":"<your account_id>", // This must be your account id on UA contracts
  "product_listing_id":", // This value is also provided by GET /cube/microcerts.json
}
```
2. POST https://ubuntu-com-10631.demos.haus/cube/microcerts/purchase.json?test_backend=true (notice "preview":"false")
```
{
  "preview": "false",
  "account_id":"<your account_id>", // This must be your account id on UA contracts
  "product_listing_id":", // This value is also provided by GET /cube/microcerts.json
}
```

- Also if you have time, please take a quick look at the cassette files see if can catch any sensitive info that should be removed (passwords, secrets, client ids, tokens etc...)

## Issue / Card

Fixes https://github.com/canonical-web-and-design/commercial-squad/issues/261
